packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -1764,6 +1764,19 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
     }
   }
 
+  function flushControlled<A>(fn: () => A): A {
```

Comment:
Should we warn if the second argument is passed? Just in case.

Also, should we reserve the space of the return value to be a thennable?